src/authenticated/onboarding/OnboardingPage.tsx

```diff
@@ -25,56 +28,47 @@ function formatMemberForMutation(updatedUser: MemberProfileFragmentFragment) {
   };
 }
 
-const MemberEditPage: React.SFC<MemberEditPageProps> = ({ history }) => {
+const MemberEditPage: React.FC<MemberEditPageProps> = ({ history }) => {
   const authContext = useContext(AuthenticationContext);
+  const updateMemberMutation = useMutation<UpdateUserMutation>(
+    UpdateUserDocument,
+    {
+      refetchQueries: ['editableUsersList']
+    }
+  );
+  const { data } = useCustomQuery<GetUserQuery>(GetUserDocument, {
+    variables: { id: authContext.authData.userId }
+  });
+  if (!data || !data.user) return null;
 
   return (
-    <UpdateUserComponent refetchQueries={['editableUsersList']}>
-      {updateMemberMutation => (
-        <GetUserComponent variables={{ id: authContext.authData.userId }}>
-          {({ loading, error, data }) => {
-            if (error) return `Error! ${error.message}`;
-            if (loading || !data || !data.user) return <LoadingIndicator />;
-
-            return (
-              <Formik
-                initialValues={data.user}
-                onSubmit={async (updatedMember, actions) => {
-                  try {
-                    await updateMemberMutation({
-                      variables: formatMemberForMutation({
-                        ...updatedMember,
-                        hasCompletedWizard: true
-                      })
-                    });
-                    history.push('/');
-                  } catch (err) {
-                    console.error('submitting error', err);
-                    actions.setSubmitting(false);
-                    alert(err);
-                  }
-                }}
-                render={({
-                  values,
-                  dirty,
-                  handleChange,
-                  handleSubmit,
-                  isSubmitting
-                }) => (
-                  <Form>
-                    <Onboarding
-                      user={values}
-                      onChange={handleChange}
-                      onSubmit={handleSubmit}
-                    />
-                  </Form>
-                )}
-              />
-            );
-          }}
-        </GetUserComponent>
+    <Formik
+      initialValues={data.user}
+      onSubmit={async (updatedMember, actions) => {
```

Comment:
I've seen this a few times too. It would be cleaner to handle the logic outside of `onSubmit` in a separate function, `handleSubmit` for instance. I consider this a non-blocker however.

Reply:
I think there might have been some typescript annoyance where the types for the function needed to be explicitly defined if pulled out of this inline context. Would be worth investigating again at some point